issues/issue320_test.cc

```diff
@@ -0,0 +1,130 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <map>
+#include <vector>
+#include <memory>
+
+#include <math.h>
+
+#include <leveldb/db.h>
+#include <leveldb/write_batch.h>
+
+using namespace std;
+
+unsigned int random(unsigned int max) {
+  return rand() % max;
+}
+
+string newString(int32_t index) {
+  const unsigned int len = 1024;
+  char bytes[len];
+  unsigned int i = 0;
+  while (i < 8) {
+    bytes[i] = 'a' + ((index >> (4 * i)) & 0xf);
+    ++i;
+  }
+  while (i < sizeof(bytes)) {
+    bytes[i] = 'a' + random(26);
+    ++i;
+  }
+  return string(bytes, sizeof(bytes));
+}
+
+int main(int argc, char** argv) {
```

Comment:
This test is fine, but there were two problems I noticed when running on the Google internal infrastructure. 1. Our tests detected leaking snapshots, and 2) We cannot create local files - they need to be put in a temporary directory. I made those two changes as well as refactoring this test to use testutil.h - which is similar to googletest.

I uploaded my refactoring to https://github.com/cmumford/leveldb/commit/4b72cb14f8da2aab12451c24b8e205aff686e9dc. You're welcome to copy my version of the file over if you want.